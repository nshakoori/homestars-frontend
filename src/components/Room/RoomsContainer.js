import React, { Component }  from 'react'
import { connect } from 'react-redux';
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { fetchRooms } from './RoomsActions';
import Error from '../errors/Error'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 40%
`
const Button = styled.button`
  font-size: 18px;

  background-color: white;
  border: solid 1px;
  margin: 1rem 0;
  padding: 1rem;
` 

class RoomsContainer extends Component {

  componentDidMount() {
    this.props.fetchRooms()
  }

  render() {
    const { rooms, isFetching, error } = this.props

    if(isFetching){
      return <p>Loading Available Rooms ...</p>
    } else if(error)
    {
      return <Error error={error} />
    }

    return(
      <Container>
        { 
          rooms.length > 0 ?
            rooms.map(room => {
              return <Button key={room.name}>{room.name}</Button>
            }) 
          :
          <p>No Rooms Available At This Time</p>
        }
      </Container>
    )
  }
}

const mapStateToProps = ({ rooms }) => {

  return { 
    isFetching: rooms.isFetching,
    rooms: rooms.rooms,
    error: rooms.error
  }
}

const mapDispatchToProps = { fetchRooms }

RoomsContainer.propTypes = {
  isFetching: PropTypes.bool,
  rooms: PropTypes.array,
  fetchRooms: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(RoomsContainer);