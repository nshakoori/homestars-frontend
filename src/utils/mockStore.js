
import configureStore from 'redux-mock-store';

const mockConfigureStore = configureStore();

const mockStore = initialState => {
    return mockConfigureStore(initialState);
}

export default mockStore;