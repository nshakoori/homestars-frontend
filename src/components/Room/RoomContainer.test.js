//import configureStore from 'redux-mock-store';
import RoomsContainer from './RoomsContainer'
import toJson from 'enzyme-to-json';
import mockStore from '../../utils/mockStore'

describe('RoomsContainer', () => {

  it('should render loading message', () => {
    const initialState = {
      isFetching: true,
      error: undefined,
      rooms: []
    }
    const store = mockStore(initialState)
    const wrapper = shallow(
      <RoomsContainer store={store}/>
    ).dive()
  
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('should render loading message', () => {
    const initialState = {
      isFetching: false,
      error: "error message",
      rooms: []
    }
    const store = mockStore(initialState)
    const wrapper = shallow(
      <RoomsContainer store={store}/>
    ).dive()
  
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('should render loading message', () => {
    const initialState = {
      isFetching: false,
      error: undefined,
      rooms: [ { id: 1, name: 'Room1'}, { id: 2, name: 'Room2'} ]
    }
    const store = mockStore(initialState)
    const wrapper = shallow(
      <RoomsContainer store={store}/>
    ).dive()
  
    expect(toJson(wrapper)).toMatchSnapshot();
  })

})