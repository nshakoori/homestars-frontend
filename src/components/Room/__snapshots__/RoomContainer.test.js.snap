// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomsContainer should render loading message 1`] = `
<RoomsContainer
  fetchRooms={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`RoomsContainer should render loading message 2`] = `
<RoomsContainer
  fetchRooms={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`RoomsContainer should render loading message 3`] = `
<RoomsContainer
  fetchRooms={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
